src/storage/txn/commands.rs

```diff
@@ -0,0 +1,957 @@
+            keys,
+            req.get_start_version().into(),
+            req.get_for_update_ts()
+                .try_into()
+                .expect("Zero for_update_ts in pessimistic transaction"),
+            req.take_context(),
+        )
+    }
+}
+
+impl From<TxnHeartBeatRequest> for TypedCommand<TxnStatus> {
+    fn from(mut req: TxnHeartBeatRequest) -> Self {
+        TxnHeartBeat::new(
+            Key::from_raw(req.get_primary_lock()),
+            req.get_start_version().into(),
+            req.get_advise_lock_ttl(),
+            req.take_context(),
+        )
+    }
+}
+
+impl From<CheckTxnStatusRequest> for TypedCommand<TxnStatus> {
+    fn from(mut req: CheckTxnStatusRequest) -> Self {
+        CheckTxnStatus::new(
+            Key::from_raw(req.get_primary_key()),
+            req.get_lock_ts().into(),
+            req.get_caller_start_ts().into(),
+            req.get_current_ts().into(),
+            req.get_rollback_if_not_exist(),
+            req.take_context(),
+        )
+    }
+}
+
+impl From<ScanLockRequest> for TypedCommand<Vec<LockInfo>> {
+    fn from(mut req: ScanLockRequest) -> Self {
+        ScanLock::new(
+            req.get_max_version().into(),
+            &req.take_start_key(),
+            req.get_limit() as usize,
+            req.take_context(),
+        )
+    }
+}
+
+impl From<ResolveLockRequest> for TypedCommand<()> {
+    fn from(mut req: ResolveLockRequest) -> Self {
+        let resolve_keys: Vec<Key> = req
+            .get_keys()
+            .iter()
+            .map(|key| Key::from_raw(key))
+            .collect();
+        let txn_status = if req.get_start_version() > 0 {
+            HashMap::from_iter(iter::once((
+                req.get_start_version().into(),
+                req.get_commit_version().into(),
+            )))
+        } else {
+            HashMap::from_iter(
+                req.take_txn_infos()
+                    .into_iter()
+                    .map(|info| (info.txn.into(), info.status.into())),
+            )
+        };
+
+        if resolve_keys.is_empty() {
+            ResolveLock::new(txn_status, None, vec![], req.take_context()).into()
+        } else {
+            let start_ts: TimeStamp = req.get_start_version().into();
+            assert!(!start_ts.is_zero());
+            let commit_ts = req.get_commit_version().into();
+            ResolveLockLite::new(start_ts, commit_ts, resolve_keys, req.take_context())
+        }
+    }
+}
+
+impl From<MvccGetByKeyRequest> for TypedCommand<MvccInfo> {
+    fn from(mut req: MvccGetByKeyRequest) -> Self {
+        MvccByKey::new(Key::from_raw(req.get_key()), req.take_context())
+    }
+}
+
+impl From<MvccGetByStartTsRequest> for TypedCommand<Option<(Key, MvccInfo)>> {
+    fn from(mut req: MvccGetByStartTsRequest) -> Self {
+        MvccByStartTs::new(req.get_start_ts().into(), req.take_context())
+    }
+}
+
+macro_rules! command {
+    (
+        $(#[$outer_doc: meta])*
+        $cmd: ident : $cmd_ty: ty {
```

Comment:
In such a macro, I prefer to use `=>` or `->` rather than `:` to express the `return` meaning.